frame/support/src/storage/mod.rs

```diff
@@ -781,6 +875,36 @@ pub struct KeyPrefixIterator<T> {
 }
 
 impl<T> KeyPrefixIterator<T> {
+	/// Creates a new `KeyPrefixIterator`, starting from `previous_key` and filtering out keys that
+	/// are not prefixed with `prefix`.
+	///
+	/// A `decode_fn` function must also be supplied, and it takes in a `&[u8]` parameter, returning
+	/// a `Result` containing the decoded key type `T` if successful, and a `codec::Error` on
+	/// failure. The `&[u8]` argument represents the raw, undecoded key without the prefix of the
+	/// current item.
+	pub fn new(
+		prefix: Vec<u8>,
+		previous_key: Vec<u8>,
+		decode_fn: fn(&[u8]) -> Result<T, codec::Error>,
+	) -> Self {
```

Comment:
I feel like better than exposing all the functions `partial_key_decode` ... we can do a better API.
Currently it is a bit error prone considering user have to create iterator with the correct decode_fn.
Instead maybe we can create a function: `start_iterating_after` (or a better name) which would set the `previous_key` to iterate after.
Then people can just do `MyStorageMap::iter().start_iterating_after(the old raw key)`
Or maybe even better `MyStorageMap::iter_after(my_key_not_raw)`.

what do you think ?

Reply:
I feel like just adding a function `set_previous_key` is more handy as you don't need to create a PrefixIterator and give the correct decode_fn.
Anyhow we can keep the `

Reply:
I totally agree with @thiolliere here. My expectation of the API was that the user stores a key, and next to `StorageMap::iter()`, they can do `StorageMap::iter_from(stored_key)`.

Reply:
Yeah, this is definitely a better idea, I just couldn't figure out how to create a brand new `PrefixIterator` and start iterating from a given key since it required a decode_fn.